website/src/pages/tech-specs.md

```diff
@@ -0,0 +1,350 @@
+  | rollback      | Rollback denotes that the changes made by the corresponding commit/delta commit was unsuccessful & hence rolled back, removing any partial files produced during such a write |
+  | savepoint     | Savepoint is a special marker to ensure a particular commit is not automatically cleaned. It helps restore the table to a point on the timeline, in case of disaster/data recovery scenarios |
+  | restore       | Restore denotes that a particular Savepoint was restored     |
+  | clean         | Maintenance activity that cleans up versions of data files that no longer will be accessed |
+  | compaction    | Maintenance to optimize the storage for query performance. This action applies the batched up updates from `deltacommit` and re-optimizes data files for query performance |
+  | replacecommit | Maintenance activity to cluster the data for better query performance. This action is different from a `commit` in that the the table state before and after are logically equivalent |
+  | indexing      | [todo]                                                       |
+  | schemacommit  | [todo]                                                       |
+
+  **Action state** - Denotes the state transition identifier (requested -> inflight -> completed)
+
+
+
+Meta-files with completed transaction state contains details about the transaction such as the number of inserts/updates/deletes per file ID, file size, and some extra metadata such as checkpoint and schema for the batch of records written. The data is written in Json and the Avro schema for each of these actions are as follows
+
+- `commit ` - [HoodieCommitMetadata](https://github.com/apache/hudi/blob/master/hudi-common/src/main/avro/HoodieCommitMetadata.avsc)
+- `deltacommit` -  [HoodieCommitMetadata](https://github.com/apache/hudi/blob/master/hudi-common/src/main/avro/HoodieCommitMetadata.avsc)
+- `rollback`- [HoodieRollbackMetadata](https://github.com/apache/hudi/blob/master/hudi-common/src/main/avro/HoodieRollbackMetadata.avsc)
+- `savepoint` - [HoodieSavepointMetadata](https://github.com/apache/hudi/blob/master/hudi-common/src/main/avro/HoodieSavePointMetadata.avsc)
+- `restore ` - [HoodieRestoreMetadata](https://github.com/apache/hudi/blob/master/hudi-common/src/main/avro/HoodieRestoreMetadata.avsc)
+- `clean`  - [HoodieCleanMetadata](https://github.com/apache/hudi/blob/master/hudi-common/src/main/avro/HoodieCleanMetadata.avsc) 
+- `compaction` - [HoodieCompactionMetadata](https://github.com/apache/hudi/blob/master/hudi-common/src/main/avro/HoodieCompactionMetadata.avsc)
+- `replacecommit` - [HoodieReplaceCommitMetadata](https://github.com/apache/hudi/blob/master/hudi-common/src/main/avro/HoodieReplaceCommitMetadata.avsc)
+- `indexing` - [HoodieIndexCommitMetadata](https://github.com/apache/hudi/blob/master/hudi-common/src/main/avro/HoodieIndexCommitMetadata.avsc)
+- `schemacommit`- 
```

Comment:
This is just `InternalSchema` serialized to byte[].
https://github.com/apache/hudi/blob/master/hudi-common/src/main/java/org/apache/hudi/internal/schema/InternalSchema.java